weepickle-implicits/src/main/scala-3/com/rallyhealth/weepickle/v1/implicits/macros.scala

```diff
@@ -0,0 +1,181 @@
+package com.rallyhealth.weepickle.v1.implicits.macros
+
+import scala.quoted.{ given, _ }
+import deriving._, compiletime._
+
+inline def getDefaultParams[T]: String => Option[() => AnyRef] = ${ getDefaultParmasImpl[T] }
+def getDefaultParmasImpl[T](using Quotes, Type[T]): Expr[String => Option[() => AnyRef]] =
```

Comment:
Minor detail (currently catching up), but while I notice it: `getDefaultParmasImpl` -> `getDefaultParamsImpl `.